editor/utils/darkMode.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+import { darkModeLocalStorageKey } from "@/components/DarkModeToggle";
+
+/**
+ * Custom hook that returns whether the app is in dark mode
+ * and uses event listener to listen for updates
+ */
+export const useCheckDarkMode = () => {
```

Comment:
When i first tried to implement this feature, the map would load the correct basemap color on first render, but if i toggled Dark mode while having the map open, the basemap wouldnt update. This useEffect paired with the event listener allows the map component to keep the app theme state up to date with when the user toggles dark mode.

It can also be used elsewhere in the app as needed -- i thought about using it [here in SidebarLayout.tsx](https://github.com/cityofaustin/vision-zero/blob/63adb99f91fe99eb78bf4d256f9b6c4d941f8dd1/editor/components/SidebarLayout.tsx#L49) but the useEffect there also sets the document attribute, so idk :thinking:

Reply:
This is really cool! Having proper dark mode state is going to be super handy for other places where we want to get fancy with our light/dark mode UI. 

One thing that jumps out at me is that it probably makes sense to move this hook into a global context (e.g. `ThemeProvider`) so that we don't duplicate this event handling if we add the hook to multiple components (like the `SidebarLayout`). Would you mind giving that a try?